pkg/placement/placement_test.go

```diff
@@ -0,0 +1,178 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package placement
+
+import (
+	"context"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/dapr/dapr/pkg/placement/raft"
+	v1pb "github.com/dapr/dapr/pkg/proto/placement/v1"
+	"github.com/phayes/freeport"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+)
+
+var testRaftServer *raft.Server
+
+// TestMain is executed only one time in the entire package to
+// start test raft server.
+func TestMain(m *testing.M) {
+	testRaftServer = raft.New("testnode", true, true, []raft.PeerInfo{
+		{
+			ID:      "testnode",
+			Address: "127.0.0.1:6060",
+		},
+	})
+
+	testRaftServer.StartRaft(nil)
+
+	for {
+		if testRaftServer.IsLeader() {
```

Comment:
is this added to make sure that the server starts up correctly? Please add comment if so.

Reply:
yes this is to wait until raft framework selects the current test raft server as leader.

Reply:
added comment